File: backend/src/main/java/arbremojo/backend/model/CourseList.java
package arbremojo.backend.model;

import arbremojo.backend.model.user.Professor;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "course_list")
@Setter
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class CourseList {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "course_list_id", nullable = false)
    private Integer courseListId;

    @JoinColumn(name = "professor_id", nullable = false)
    private Integer professorId;
    @JoinColumn(name = "course_id", nullable = false)
    private Integer courseId;
}

File: frontend/src/model/course-list.ts
export class CourseList {
  courseListId: string;
  professorId: string;
  courseId: string;

  constructor(courseListId: string, professorId: string, courseId: string) {
    this.courseListId = courseListId;
    this.professorId = professorId;
    this.courseId = courseId;
  }
}
